airbyte-integrations/connectors/source-spacex-api/source_spacex_api/spacex_api.yaml

```diff
@@ -0,0 +1,39 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+
+  requester:
+    url_base: "https://api.spacexdata.com/v4/"
+    http_method: "GET"
+    authenticator:
+      type: ApiKeyAuthenticator
+      header: "apikey"
+      api_token: "{{ config['api_key'] }}"
+
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+    requester:
+      $ref: "*ref(definitions.requester)"
+
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+
+  launches_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "launches"
+      primary_key: "id"
+      path: "/launches/{{config['id'] or 'latest'}}"
+
+streams:
+  - "*ref(definitions.launches_stream)"
```

Comment:
@btkcodedev are you going to ingest data from other endpoints? like crew, rockets, etc?
What not use the `all` endpoint instead of latest?

Reply:
@btkcodedev please check this comment.

Reply:
@marcosmarxm Yes. But all endpoints is having some issues with acceptance test. Thus the config option is provided to select the required option
Ex: For launches stream, Latest, past or upcoming could be the options :)
```
  launches_stream:
    $ref: "*ref(definitions.base_stream)"
    $options:
      name: "launches"
      primary_key: "id"
      path: "/launches/{{config['id'] or config['options'] or latest}}"
```

Reply:
Thus comment resolved :)